AI/Nullkiller/Analyzers/DangerHitMapAnalyzer.cpp

```diff
@@ -165,7 +165,7 @@ void DangerHitMapAnalyzer::calculateTileOwners()
 	auto addTownHero = [&](const CGTownInstance * town)
 	{
 			auto townHero = temporaryHeroes.emplace_back(std::make_unique<CGHeroInstance>(town->cb)).get();
-			CRandomGenerator rng;
+			CRandomGenerator & rng = CRandomGenerator::getDefault();
```

Comment:
To be honest this is not much better. Not worse, sure, but not really an improvement.
`getDefault` returns thread-local generator and AI shuffles through threads constantly.

Reply:
Now that you mention it. It is _worse_, I believe. `CRandomGenerator` uses `std::minstd_rand` under the hood and passing it by reference between threads is not safe (please correct me if I am wrong). Concurrently generating numbers might lead to UB due to concurrent RNG state modification. With that said, I believe there's some more work to be done here :/